tools/clang_tidy.py

```diff
@@ -8,28 +8,35 @@
 import sys
 
 
-DEFAULT_FILE_PATTERN = r".*\.[ch](pp)?"
+DEFAULT_FILE_PATTERN = r".*\.c(c|pp)?"
 
 # @@ -start,count +start,count @@
 CHUNK_PATTERN = r"^@@\s+-\d+,\d+\s+\+(\d+)(?:,(\d+))?\s+@@"
 
 
-def run_shell_command(arguments, process_name=None):
+def run_shell_command(arguments, process_name=None, verbose=False):
     """Executes a shell command."""
     assert len(arguments) > 0
     try:
-        output = subprocess.check_output(arguments, stderr=subprocess.STDOUT)
-    except OSError:
-        _, e, _ = sys.exc_info()
+        if verbose:
```

Comment:
I don't understand. Don't I *always* want to see stderr? Why would I throw all that information away by default.

Reply:
Because it's super noisy in CI. I want people to immediately see what lines they need to fix in their code, and not have to first look over 50 lines of clang-tidy verbose output

Reply:
Can you show what it looks like with and without verbose?

Reply:
My opinion on this sort of thing is unless you're sure the information filtering is 100% correct, you're bound to throw out useful information, and so when something breaks (and things always break), the person debugging will be wondering what is going on, and they're not going to be able to see the relevant information from the logs because you hid it.

(Of course, all of this is written under the assumption that clang-tidy's default output is of reasonable verbosity and not, like, debug level verbosity. Which, to be fair, I don't know! I haven't seen it before.)

Reply:
OK, so I played around with clang-tidy a bit, and I guess what your problem is, is that clang-tidy will report if it fails to compile a file, but that doesn't count as an actual error, it will still return exit code 0 if that's the only problem. But in that case, isn't it better to error in this case? Otherwise, someone can make a build system change which breaks clang-tidy, and you'll lose all of your tidying *silently*, just because clang-tidy no longer understand anymore.